libs/@local/harpc/net/src/session/error.rs

```diff
@@ -1,72 +1,126 @@
-use bytes::Bytes;
-use harpc_wire_protocol::response::kind::ErrorCode;
+use core::error::{Error, Request};
 
-use crate::codec::WireError;
+use harpc_codec::error::ErrorCode;
 
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, thiserror::Error)]
 #[error("The session layer has encountered an error, the connection has been closed")]
 pub struct SessionError;
 
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct TransactionError {
-    pub code: ErrorCode,
-    pub bytes: Bytes,
-}
-
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, thiserror::Error)]
-#[error(
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    derive_more::Display,
+    serde::Serialize,
+    serde::Deserialize,
```

Comment:
Do we want to import the names instead? 😄 
```suggestion
    Display,
    Serialize,
    Deserialize,
```

Motivation: All of these traits are super common and very well known. It doesn't matter where the derive comes from (same for `Error`).

Reply:
I personally like it more non-std macros use the crate they're coming from, it allows the reader to see: this is an std derive / macro or crate macro vs. this is an external macro. I can change it ofc, but that's just my personal preference (one example would also be `derive_more::Error` vs. `thiserror::Error`, do slightly different things). I can also change it, but then we'd need to change it in both harpc and hql I believe.

Reply:
I think consistency at the code base would be good and I haven't looked at this before so I don't think it's worth partially doing one or the other. I see your argument!